In [5]:
import pandas as pd 
from bs4 import BeautifulSoup
import requests


# Finalish


In [200]:
# setting up df
executive_orders = pd.DataFrame(columns=['order_no', 'title', 'text', 'date'])

#making the scraping into a function
def scrape_executive_orders(start, end):
    for num in range(start, end):
        url = f"https://en.wikisource.org/wiki/Executive_Order_{num}"
        order_no = num
        page = requests.get(url)
        soup = BeautifulSoup(page.text, 'html.parser')
        if soup.find('span', {'id': 'Wikisource_does_not_have_a_text_with_this_exact_name'}):
            continue
        else:
            title = soup.find(id='header_section_text').text
            date = soup.find('span', class_='plainlinks').text.split(' ')[-3:]
            date_string = ' '.join(date)
            
            # Find all paragraph tags
            texts = soup.find_all('p')

            # Create string with all the text that stops before copyright stuff
            stop_text1 = "This work is in the public domain"
            stop_text2 = "Public domain"
            full_text = ''
            skip_keywords = ['from:']
            for text in texts:
                if "<b>" not in str(text):
                    paragraph_text = text.get_text().lower()
                    if any(keyword in paragraph_text for keyword in skip_keywords):
                        continue
                    if stop_text1 in paragraph_text or stop_text2 in paragraph_text:
                        break
                    full_text += paragraph_text
                    full_text = full_text.replace('\n', ' ')

            #add to dataframe
            new_row = {'order_no': order_no, 'title': title, 'text': full_text, 'date': date_string}
            executive_orders.loc[len(executive_orders)] = new_row
    return executive_orders

In [201]:
scrape_executive_orders(6071, 7000)
executive_orders

order_no                                              title  \
0       6071                 Appointment of August A. Gennerich   
1       6072  Transfer of Control and Jurisdiction of Certai...   
2       6073      Regulations Concerning the Operation of Banks   
3       6084         Consolidating Federal Farm Credit Agencies   
4       6086        Establishing the Rio Grande Wildlife Refuge   
5       6092                         Veterans Regulations No. 4   
6       6093                         Veterans Regulations No. 5   
7       6098  Veterans Regulation No.10: Miscellaneous Provi...   
8       6099  Veterans Regulation No.11: Disclosure of Infor...   
9       6100  Veterans Regulations No.12: Presumption of Ent...   
10      6101           Starting the Civilian Conservation Corps   
11      6102  Forbidding the Hoarding of Gold Coin, Gold Bul...   
12      6108  Waiver of Provisions of Executive Order of Jan...   
13      6111                On Transactions in Foreign Exchange   
14      6145                  Consolidating Government Agencies   
15      6166                 Organization of Executive Agencies   
16      6173  Executive Order 6173 on the National Recovery Act   
17      6174  Executive Order 6174 on the Public Works Admin...   
18      6182          Regarding Administration of Farm Products   
19      6198  Authorization to appoint Harold L. Ickes head ...   
20      6199  Prohibition of Transportation in Interstate an...   
21      6247  Preparation, Form, Style, and Safeguarding of ...   
22      6259  Amendment of Executive Order of January 28, 18...   
23      6260  Relating to the Hoarding, Export, and Earmarki...   
24      6261  Relating to the Sale and Export of Gold Recove...   
25      6340          Creating The Commodity Credit Corporation   
26      6355  Limiting the effect of the National Recovery A...   
27      6359   Relating to Gold Recovered from Natural Deposits   
28      6360         Authorization to Appoint Mr. Wade Crawford   
29      6443  Authorization to Modify or Make Exceptions in ...   
30      6470  Creation of the Public Works Emergency Housing...   
31      6474  Creation of the Federal Alcohol Control Admini...   
32      6497  Amendment of Executive Order No. 6247, of Augu...   
33      6511       Powers of the National Labor Board Increased   
34      6513  Adding the Members of the Special Industrial R...   
35      6514  Authorization to the Electric Home and Farm Au...   
36      6518  Waiving Provisions of Executive Order of Janua...   
37      6544  Withdrawal of Public Lands for Forest Fire Loo...   
38      6545  Change of Name of Customs Port of Nyando, New ...   
39      6556  Amendment of Executive Order No. 6260 of Augus...   
40      6558  Relating to Receipt of Gold on Consignment by ...   
41      6559  Amending the Executive Order of March 10, 1933...   
42      6560  Regulating Transactions in Foreign Exchange, T...   
43      6581      Creating The Export-Import Bank of Washington   
44      6586  Revocation of Section 18 of Executive Order No...   
45      6591                 Directing the Army to Fly the Mail   
46      6617  Withdrawal of Public Lands for Use in Connecti...   
47      6667            Withdrawal of Public Lands for Resurvey   
48      6668  Veterans Regulation No. 1(e); Entitlement to P...   
49      6669  Veterans Regulation No. 12(b); Presumption of ...   
50      6670  Transfer of Functions of the Veterans' Adminis...   
51      6680  Relating to Uniform Usage in Regard to Geograp...   
52      6715  Filing of Functional Organization Charts With ...   
53      6781   Withdrawal of Public Lands for Resurvey; Arizona   
54      6824  Waiving Provisions of Executive Order of Janua...   
55      6935  Placing Certain Islands in the Pacific Ocean U...   
56      6936  Announcing the Index Figures for the Cost of L...   
57      6967  Veterans Regulation No. 1 (f), Entitlement to ...   

                                                 t

In [202]:
executive_orders.to_csv('executive_orders.csv')

In [203]:
#7000 to 8000
scrape_executive_orders(7000, 8000)
executive_orders.to_csv('executive_orders.csv')

In [213]:
# Only need to clean up two dates
executive_orders.at[3, 'date'] = 'March 27, 1933'
executive_orders.at[4, 'date'] = 'March 28, 1933'

In [214]:
executive_orders

order_no                                              title  \
0       6071                 Appointment of August A. Gennerich   
1       6072  Transfer of Control and Jurisdiction of Certai...   
2       6073      Regulations Concerning the Operation of Banks   
3       6084         Consolidating Federal Farm Credit Agencies   
4       6086        Establishing the Rio Grande Wildlife Refuge   
..       ...                                                ...   
91      7944  Amendment of the Executive Order of January 17...   
92      7956  Authorizing the Employment Until February 1, 1...   
93      7962  Amendment of Executive Order No. 4601 of March...   
94      7983     Establishing the Breton Bird Refuge, Louisiana   
95      7998  Establishing the Interdepartmental Committee o...   

                                                 text              date  
0    august a. gennerich may be appointed an opera...     March 8, 1933  
1    canal zoneby virtue of the authority vested i...     March 8, 1933  
2    by virtue of the authority vested in me by se...    March 10, 1933  
3    whereas sections 401 and 403 of title iv of p...    March 27, 1933  
4    it is hereby ordered that executive order no....    March 28, 1933  
..                                                ...               ...  
91  by virtue of and pursuant to the authority ves...    August 4, 1938  
92  by virtue of and pursuant to the authority ves...   August 19, 1938  
93   by virtue of and pursuant to the authority ve...   August 22, 1938  
94  by virtue of and pursuant to the authority ves...   October 4, 1938  
95  by virtue of the authority vested in me as pre...  October 29, 1938  

[96 rows x 4 columns]

In [215]:
executive_orders.to_csv('executive_orders.csv')

In [216]:
#8000 to 9000
scrape_executive_orders(8000, 9000)
executive_orders.to_csv('executive_orders.csv')

In [218]:
#9000 to 10000
# if re-running, cut off before 9540, since that will have to be added manually (see below)
scrape_executive_orders(9000, 10000)
executive_orders.to_csv('executive_orders.csv')

AttributeError: 'NoneType' object has no attribute 'text'

In [219]:
executive_orders

order_no                                              title  \
0        6071                 Appointment of August A. Gennerich   
1        6072  Transfer of Control and Jurisdiction of Certai...   
2        6073      Regulations Concerning the Operation of Banks   
3        6084         Consolidating Federal Farm Credit Agencies   
4        6086        Establishing the Rio Grande Wildlife Refuge   
..        ...                                                ...   
182      9508  Authorizing the Secretary of War to Take Posse...   
183      9511  Authorizing the Secretary of War To Take Posse...   
184      9533  Changing the Name of the Customs Port of Entry...   
185      9537  Amending Executive Order 8396 of April 18, 194...   
186      9538  Authorizing Certification for Probational Appo...   

                                                  text               date  
0     august a. gennerich may be appointed an opera...      March 8, 1933  
1     canal zoneby virtue of the authority vested i...      March 8, 1933  
2     by virtue of the authority vested in me by se...     March 10, 1933  
3     whereas sections 401 and 403 of title iv of p...     March 27, 1933  
4     it is hereby ordered that executive order no....     March 28, 1933  
..                                                 ...                ...  
182  whereas the national war labor board has found...  December 27, 1944  
183  by virtue of the authority vested in me by the...   January 12, 1945  
184   by virtue of the authority vested in me by se...     March 23, 1945  
185  amending executive order 8396 of april 18, 194...     April 11, 1945  
186  by virtue of the authority vested in me by sec...     April 13, 1945  

[187 rows x 4 columns]

Got to 9538 before throwing an error. It's just EO 9540 that appears to be formatted weirdly, so adding that one in manually. https://en.wikisource.org/wiki/Executive_Order_9540

In [222]:
title = "AUTHORIZING THE PETROLEUM ADMINISTRATIOR TO TAKE THE POSESSION OF AND TO THE CITIES SERVICE REFINING CORPORATION, LOCATED IN AND AROUND LAKE CHARLES, LOUISIANA".title()
text = "Whereas after investigation I find and proclaim that the plants and facilities owned or leased by the Cities Service Refining Corporation located in and around Lake Charles, Louisiana, are equipped for the manufacture and production of articles and materials that are required for the war effort , or that are useful in connection therewith; that there are existing interruptions of the operation of the said plants and facilities as a result of a labor disturbance; that the war efforts will be unduly impeded or delayed by these interruptions; and that the exercise, as a hereinafter specified, of the powers vested in me is necessary to insure, in the interest of the war effort, the operation of these plants and facilities: Now, Therefore, by virtue of the power and authority vested in me by the Constitution and laws of the United States, including section 9 of the Selective Training and Service Act of 1940 (54 Stat. 892) as amended by the War Labor Disputes Act (57 Stat. 163), as President of the United States and Commander in Chief of the Army and Navy of the United States, it is hereby ordered as follows: 1. The Petroleum Administrator is hereby authorized and directed, through and with the aid of any persons or instrumentalities that he may designate, to take possession of the plants and facilities of the Cities Service Refining Corporation located in and around Lake Charles, Louisiana, and to the extent that he may deem necessary, of any real or personal property , and other assets wherever situated, used in connection with the operations thereof; to operate or to arrange for the operation of the plants and facilities in any manner, in accordance with applicable laws, that he may deem necessary for the successful prosecution of the war; to exercise any contractual or other rights of the Cities Service Refining Corporation, and to continue or discontinue the employment of, or to employ, any persons; and to do any other things that he may deem necessary for, or incidental to, the operation of the sale plants and facilities and the production, sale, and distribution of the products thereof; and to take any other steps that he deems necessary to carry out the provisions and purposes of this order. 2. All Federal agencies, including, but not limited to, the National War Labor Board, the War Manpower Commission the National Selective Service System, and the Department of Justice, are directed to cooperate with the Petroleum Administrator to the fullest extent possible in carrying out the purposes of this order. 3. The Petroleum Administrator shall provide such protection as may be required for such plants and facilities, and protection for all persons employed or seeking employment therein, both while at work, and for their families and homes; and upon the request of the Petroleum Administrator, or such person as may be designated to act for him, the Secretary of War shall take such action as may be necessary to provide such action as may to such persons and property. 4. The Petroleum Administrator shall permit the management of the plants and facilities taken under the provisions of this order to continue with its managerial functions to the maximum degree possible consistent with the aims of the order. 5. Possession, control, and operation of any plant or facility, or part thereof, taken under this order shall be terminated b the Petroleum Administrator after possession of such plant, facility, or part thereof, is no longer required for the successful prosecution of the war, but in no event more than sixty days after the restoration of productive efficiency thereof prevailing prior to the taking of possession thereof. 6. The Petroleum Administrator may delegate any and all power, authority, and discretion conferred upon him by this order to the Deputy Petroleum Administrator. The Petroleum Administrator and Deputy Petroleum Administrator may exercise the powers, authority, and discretion conferred upon them by this or under the provisions of the Petroleum Administration for War in such manner as the Petroleum Administrator of Deputy Petroleum Administrator may determine.".lower()
date_string = "April 17, 1945"
num = "9540"
new_row = {'order_no': num, 'title': title, 'text': full_text, 'date': date_string}
executive_orders.loc[len(executive_orders)] = new_row
executive_orders

order_no                                              title  \
0        6071                 Appointment of August A. Gennerich   
1        6072  Transfer of Control and Jurisdiction of Certai...   
2        6073      Regulations Concerning the Operation of Banks   
3        6084         Consolidating Federal Farm Credit Agencies   
4        6086        Establishing the Rio Grande Wildlife Refuge   
..        ...                                                ...   
183      9511  Authorizing the Secretary of War To Take Posse...   
184      9533  Changing the Name of the Customs Port of Entry...   
185      9537  Amending Executive Order 8396 of April 18, 194...   
186      9538  Authorizing Certification for Probational Appo...   
187      9540  Authorizing The Petroleum Administratior To Ta...   

                                                  text              date  
0     august a. gennerich may be appointed an opera...     March 8, 1933  
1     canal zoneby virtue of the authority vested i...     March 8, 1933  
2     by virtue of the authority vested in me by se...    March 10, 1933  
3     whereas sections 401 and 403 of title iv of p...    March 27, 1933  
4     it is hereby ordered that executive order no....    March 28, 1933  
..                                                 ...               ...  
183  by virtue of the authority vested in me by the...  January 12, 1945  
184   by virtue of the authority vested in me by se...    March 23, 1945  
185  amending executive order 8396 of april 18, 194...    April 11, 1945  
186  by virtue of the authority vested in me by sec...    April 13, 1945  
187  by virtue of the authority vested in me as pre...    April 17, 1945  

[188 rows x 4 columns]

In [223]:
# continuing to 10000
scrape_executive_orders(9541, 10000)
executive_orders.to_csv('executive_orders.csv')

AttributeError: 'NoneType' object has no attribute 'text'

In [224]:
executive_orders

order_no                                              title  \
0        6071                 Appointment of August A. Gennerich   
1        6072  Transfer of Control and Jurisdiction of Certai...   
2        6073      Regulations Concerning the Operation of Banks   
3        6084         Consolidating Federal Farm Credit Agencies   
4        6086        Establishing the Rio Grande Wildlife Refuge   
..        ...                                                ...   
286      9979  Prescribing Portions of the Selective Service ...   
287      9980  Regulations Governing Fair Employment Practice...   
288      9981  Establishing the President's Committee on Equa...   
289      9982  Extending the Provisions of Executive Order No...   
290      9983  Exemption of Garland S. Ferguson From Compulso...   

                                                  text            date  
0     august a. gennerich may be appointed an opera...   March 8, 1933  
1     canal zoneby virtue of the authority vested i...   March 8, 1933  
2     by virtue of the authority vested in me by se...  March 10, 1933  
3     whereas sections 401 and 403 of title iv of p...  March 27, 1933  
4     it is hereby ordered that executive order no....  March 28, 1933  
..                                                 ...             ...  
286  by virtue of the authority vested in me by tit...   July 20, 1948  
287  whereas it is desirable and in the public inte...   July 26, 1948  
288  ⁠whereas it is essential that there be maintai...   July 26, 1948  
289  by virtue of the authority vested in me by the...   July 29, 1948  
290  executive order 9983, exempting garland s. fer...   July 31, 1948  

[291 rows x 4 columns]

Got to 9983. 9984 doesn't have the date formatted correctly. Adding manually.

In [226]:
url = "https://en.wikisource.org/wiki/Executive_Order_9984"
page = requests.get(url)
soup = BeautifulSoup(page.text, 'html.parser')
order_no = 9984
title = soup.find(id='header_section_text').text
date_string = 'July 31, 1948' #hard coding

# Find all paragraph tags
texts = soup.find_all('p')

# Create string with all the text that stops before copyright stuff
stop_text1 = "This work is in the public domain"
stop_text2 = "Public domain"
full_text = ''
skip_keywords = ['from:']
for text in texts:
    if "<b>" not in str(text):
        paragraph_text = text.get_text().lower()
        if any(keyword in paragraph_text for keyword in skip_keywords):
            continue
        if stop_text1 in paragraph_text or stop_text2 in paragraph_text:
            break
        full_text += paragraph_text
        full_text = full_text.replace('\n', ' ').split(' ')[:-9] #removing last 9 words
        full_text = ' '.join(full_text)

#add to dataframe
new_row = {'order_no': order_no, 'title': title, 'text': full_text, 'date': date_string}
executive_orders.loc[len(executive_orders)] = new_row
executive_orders

order_no                                              title  \
0        6071                 Appointment of August A. Gennerich   
1        6072  Transfer of Control and Jurisdiction of Certai...   
2        6073      Regulations Concerning the Operation of Banks   
3        6084         Consolidating Federal Farm Credit Agencies   
4        6086        Establishing the Rio Grande Wildlife Refuge   
..        ...                                                ...   
287      9980  Regulations Governing Fair Employment Practice...   
288      9981  Establishing the President's Committee on Equa...   
289      9982  Extending the Provisions of Executive Order No...   
290      9983  Exemption of Garland S. Ferguson From Compulso...   
291      9984  Regulations Governing the Furnishing of Clothi...   

                                                  text            date  
0     august a. gennerich may be appointed an opera...   March 8, 1933  
1     canal zoneby virtue of the authority vested i...   March 8, 1933  
2     by virtue of the authority vested in me by se...  March 10, 1933  
3     whereas sections 401 and 403 of title iv of p...  March 27, 1933  
4     it is hereby ordered that executive order no....  March 28, 1933  
..                                                 ...             ...  
287  whereas it is desirable and in the public inte...   July 26, 1948  
288  ⁠whereas it is essential that there be maintai...   July 26, 1948  
289  by virtue of the authority vested in me by the...   July 29, 1948  
290  executive order 9983, exempting garland s. fer...   July 31, 1948  
291  by virtue of and pursuant to the authority ves...   July 31, 1948  

[292 rows x 4 columns]

In [227]:
# finishing to 10000
scrape_executive_orders(9985, 10000)
executive_orders.to_csv('executive_orders.csv')

In [228]:
# adjusting the function for if there is weird formatting
# will return None for date/title if not found

#making the scraping into a function
def scrape_executive_orders(start, end):
    for num in range(start, end):
        url = f"https://en.wikisource.org/wiki/Executive_Order_{num}"
        order_no = num
        page = requests.get(url)
        soup = BeautifulSoup(page.text, 'html.parser')
        if soup.find('span', {'id': 'Wikisource_does_not_have_a_text_with_this_exact_name'}):
            continue
        else:

            # title
            title_element = soup.find(id='header_section_text')
            if title_element:
                title = title_element.text
            else:
                title = None


            date_element = soup.find('span', class_='plainlinks')
            if date_element:
                date = date_element.text.split(' ')[-3:]
                date_string = ' '.join(date)
            else:
                date_string = None
            
            # Find all paragraph tags
            texts = soup.find_all('p')

            # Create string with all the text that stops before copyright stuff
            stop_text1 = "This work is in the public domain"
            stop_text2 = "Public domain"
            full_text = ''
            skip_keywords = ['from:']
            for text in texts:
                if "<b>" not in str(text):
                    paragraph_text = text.get_text().lower()
                    if any(keyword in paragraph_text for keyword in skip_keywords):
                        continue
                    if stop_text1 in paragraph_text or stop_text2 in paragraph_text:
                        break
                    full_text += paragraph_text
                    full_text = full_text.replace('\n', ' ')

            #add to dataframe
            new_row = {'order_no': order_no, 'title': title, 'text': full_text, 'date': date_string}
            executive_orders.loc[len(executive_orders)] = new_row
    return executive_orders

In [229]:
#10000 to 11000
scrape_executive_orders(10000, 11000)

order_no                                              title  \
0        6071                 Appointment of August A. Gennerich   
1        6072  Transfer of Control and Jurisdiction of Certai...   
2        6073      Regulations Concerning the Operation of Banks   
3        6084         Consolidating Federal Farm Credit Agencies   
4        6086        Establishing the Rio Grande Wildlife Refuge   
..        ...                                                ...   
474     10995  Assigning Telecommunications Management Functions   
475     10996  Promulgating Regulations Concerning Withholdin...   
476     10997  Assigning Emergency Preparedness Functions to ...   
477     10998  Assigning Emergency Preparedness Functions to ...   
478     10999  Assigning Emergency Preparedness Functions to ...   

                                                  text               date  
0     august a. gennerich may be appointed an opera...      March 8, 1933  
1     canal zoneby virtue of the authority vested i...      March 8, 1933  
2     by virtue of the authority vested in me by se...     March 10, 1933  
3     whereas sections 401 and 403 of title iv of p...     March 27, 1933  
4     it is hereby ordered that executive order no....     March 28, 1933  
..                                                 ...                ...  
474   whereas telecommunications is vital to the se...  February 16, 1962  
475   by virtue of the authority vested in me by th...  February 16, 1962  
476   by virtue of the authority vested in me as pr...  February 16, 1962  
477   by virtue of the authority vested in me as pr...  February 16, 1962  
478   by virtue of the authority vested in me as pr...  February 16, 1962  

[479 rows x 4 columns]

In [230]:
executive_orders.to_csv('executive_orders.csv')

In [231]:
# 11000 to 12000
scrape_executive_orders(11000, 12000)
executive_orders.to_csv('executive_orders.csv')

In [232]:
executive_orders

order_no                                              title  \
0        6071                 Appointment of August A. Gennerich   
1        6072  Transfer of Control and Jurisdiction of Certai...   
2        6073      Regulations Concerning the Operation of Banks   
3        6084         Consolidating Federal Farm Credit Agencies   
4        6086        Establishing the Rio Grande Wildlife Refuge   
..        ...                                                ...   
589     11957  Designation of Certain Officers to Act as Secr...   
590     11965        Establishing the Humanitarian Service Medal   
591     11967  Relating to violations of the Selective Servic...   
592     11969  Administration of the Emergency Natural Gas Ac...   
593     11975                  President's Economic Policy Board   

                                                  text              date  
0     august a. gennerich may be appointed an opera...     March 8, 1933  
1     canal zoneby virtue of the authority vested i...     March 8, 1933  
2     by virtue of the authority vested in me by se...    March 10, 1933  
3     whereas sections 401 and 403 of title iv of p...    March 27, 1933  
4     it is hereby ordered that executive order no....    March 28, 1933  
..                                                 ...               ...  
589   by virtue of the authority vested in me by se...  January 13, 1977  
590   by virtue of the authority vested in me as pr...  January 19, 1977  
591  the following actions shall be taken to facili...  January 21, 1977  
592  by virtue of the authority vested in me by the...  February 2, 1977  
593  by virtue of the authority vested in me by the...     March 7, 1977  

[594 rows x 4 columns]

In [233]:
# 12000 to 13000
scrape_executive_orders(12000, 13000)
executive_orders.to_csv('executive_orders.csv')

In [234]:
executive_orders

order_no                                              title  \
0         6071                 Appointment of August A. Gennerich   
1         6072  Transfer of Control and Jurisdiction of Certai...   
2         6073      Regulations Concerning the Operation of Banks   
3         6084         Consolidating Federal Farm Credit Agencies   
4         6086        Establishing the Rio Grande Wildlife Refuge   
...        ...                                                ...   
1350     12995             Amendment to Executive Order No. 12873   
1351     12996  Management and General Public Use of the Natio...   
1352     12997   Korean Peninsula Energy Development Organization   
1353     12998             Amendment to Executive Order No. 11880   
1354     12999  Educational Technology: Ensuring Opportunity f...   

                                                   text            date  
0      august a. gennerich may be appointed an opera...   March 8, 1933  
1      canal zoneby virtue of the authority vested i...   March 8, 1933  
2      by virtue of the authority vested in me by se...  March 10, 1933  
3      whereas sections 401 and 403 of title iv of p...  March 27, 1933  
4      it is hereby ordered that executive order no....  March 28, 1933  
...                                                 ...             ...  
1350  by the authority vested in me as president by ...  March 25, 1996  
1351   by the authority vested in me as president by...  March 25, 1996  
1352  by the authority vested in me as president by ...   April 1, 1996  
1353   by virtue of the authority vested in me as pr...   April 5, 1996  
1354  in order to ensure that american children have...  April 17, 1996  

[1355 rows x 4 columns]

In [235]:
# 13000 to 14000
scrape_executive_orders(13000, 14000)
executive_orders.to_csv('executive_orders.csv')

In [237]:
executive_orders

order_no                                              title  \
0         6071                 Appointment of August A. Gennerich   
1         6072  Transfer of Control and Jurisdiction of Certai...   
2         6073      Regulations Concerning the Operation of Banks   
3         6084         Consolidating Federal Farm Credit Agencies   
4         6086        Establishing the Rio Grande Wildlife Refuge   
...        ...                                                ...   
2181     13995  Ensuring an Equitable Pandemic Response and Re...   
2182     13996  Establishing the COVID-19 Pandemic Testing Boa...   
2183     13997  Improving and Expanding Access to Care and Tre...   
2184     13998  Promoting COVID-19 Safety in Domestic and Inte...   
2185     13999                Protecting Worker Health and Safety   

                                                   text              date  
0      august a. gennerich may be appointed an opera...     March 8, 1933  
1      canal zoneby virtue of the authority vested i...     March 8, 1933  
2      by virtue of the authority vested in me by se...    March 10, 1933  
3      whereas sections 401 and 403 of title iv of p...    March 27, 1933  
4      it is hereby ordered that executive order no....    March 28, 1933  
...                                                 ...               ...  
2181  by the authority vested in me as president by ...  January 21, 2021  
2182  by the authority vested in me as president by ...  January 21, 2021  
2183  by the authority vested in me as president by ...  January 21, 2021  
2184  by the authority vested in me as president by ...  January 21, 2021  
2185  by the authority vested in me as president by ...  January 21, 2021  

[2186 rows x 4 columns]

In [238]:
# 14000 to 14111 -- the most recent EO
scrape_executive_orders(14000, 14111)
executive_orders.to_csv('executive_orders.csv')

In [239]:
executive_orders

order_no                                              title  \
0         6071                 Appointment of August A. Gennerich   
1         6072  Transfer of Control and Jurisdiction of Certai...   
2         6073      Regulations Concerning the Operation of Banks   
3         6084         Consolidating Federal Farm Credit Agencies   
4         6086        Establishing the Rio Grande Wildlife Refuge   
...        ...                                                ...   
2248     14073  Enhancing the National Quantum Initiative Advi...   
2249     14074                                               None   
2250     14075                                               None   
2251     14092                                               None   
2252     14110  Safe, Secure, and Trustworthy Development and ...   

                                                   text              date  
0      august a. gennerich may be appointed an opera...     March 8, 1933  
1      canal zoneby virtue of the authority vested i...     March 8, 1933  
2      by virtue of the authority vested in me by se...    March 10, 1933  
3      whereas sections 401 and 403 of title iv of p...    March 27, 1933  
4      it is hereby ordered that executive order no....    March 28, 1933  
...                                                 ...               ...  
2248  by the authority vested in me as president by ...       May 4, 2022  
2249   by the authority vested in me as president by...      May 25, 2022  
2250  by the authority vested in me as president by ...     June 16, 2022  
2251   by the authority vested in me as president by...    March 14, 2023  
2252   by the authority vested in me as president by...  October 30, 2023  

[2253 rows x 4 columns]

In [249]:
#nulls:
print('Nulls in title:', executive_orders['title'].isna().sum())
print('Nulls in date:', executive_orders['date'].isna().sum())
print('Nulls in text:', executive_orders['text'].isna().sum())

Nulls in title: 14
Nulls in date: 6
Nulls in text: 0
